TestFoundation/TestNSNumberBridging.swift

```diff
@@ -515,15 +483,22 @@ class TestNSNumberBridging : XCTestCase {
                 let uint = UInt(exactly: number!)
                 XCTAssertEqual(UInt(exactly: interestingValue), uint)
 
-                let float = Float(truncating: number!)
-                let expectedFloat = interestingValue
-                testFloat(expectedFloat, float)
-
-                // FIXME: Double.nan doesn't round-trip through NSNumber
-                if !interestingValue.isNaN {
-                  let double = Double(exactly: number!)
-                  let expectedDouble = Double(exactly: interestingValue)
-                  testDouble(expectedDouble, double)
+                let float = Float(exactly: number!)
+                let expectedFloat = Float(exactly: interestingValue)
+                if interestingValue.isNaN {
+                    expectTrue(float?.isNaN == true)
+                    expectNil(expectedFloat)
+                } else {
+                    testFloat(expectedFloat, float)
+                }
+              
+                let double = Double(exactly: number!)
+                let expectedDouble = Double(exactly: interestingValue)
+                if interestingValue.isNaN {
+                    expectTrue(double?.isNaN == true)
+                    expectNil(expectedDouble)
```

Comment:
We have the XCTest functions here instead of the standard library ones.

Reply:
Thanks—had tried to type-check this in my head 😳